### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

File: commands.js [hunk 1]
```diff
@@ -272,6 +272,30 @@ exports.commands = {
 	},
 	backhelp: ["/back - Unblocks challenges and/or private messages, if either are blocked."],
 
+	rank: function (target, room, user) {
+		if (!target) target = user.name;
+
+		let buffer = '<div class="ladder"><table>';
+		buffer += '<tr><td colspan="8">User: <strong>' + Tools.escapeHTML(target) + '</strong></td></tr>';
+
+		let ratings;
```
Comment (Slayer95) [inline commands.js]: Initialize to empty string?

**Comment by Slayer95** [comment]
The ``/rank`` implementation needs some sort of feature-testing so that it doesn't crash PS main server.

**Comment by bumbadadabum** [comment]
Thanks zarel I fixed it.

File: commands.js [hunk 2]
```diff
@@ -272,6 +272,26 @@ exports.commands = {
 	},
 	backhelp: ["/back - Unblocks challenges and/or private messages, if either are blocked."],
 
+	rank: function (target, room, user) {
+		if (!target) target = user.name;
+
+		let buffer = '<div class="ladder"><table>';
+		buffer += '<tr><td colspan="8">User: <strong>' + Tools.escapeHTML(target) + '</strong></td></tr>';
```
Comment (Zarel) [inline commands.js]: Construct the buffer inside the `then` function.

Long-term buffers are a huge waste of RAM if you don't actually need them.

File: commands.js [hunk 3]
```diff
@@ -272,6 +272,26 @@ exports.commands = {
...
+
+		Ladders.Ladder.visualizeAll(target).then(values => {
```
Comment (Zarel) [inline commands.js]: You might want to export `visualizeAll` directly from `Ladders` so you don't have to do `Ladders.Ladder`.

File: commands.js [hunk 4]
```diff
@@ -272,6 +272,26 @@ exports.commands = {
...
+			let ratings = values.join('');
```
Comment (Zarel) [inline commands.js]: Temporary buffers are completely unnecessary. Do you hate RAM?

File: commands.js [hunk 5]
```diff
@@ -272,6 +272,26 @@ exports.commands = {
 	},
 	backhelp: ["/back - Unblocks challenges and/or private messages, if either are blocked."],
 
+	rank: function (target, room, user) {
+		if (!target) target = user.name;
```
Comment (Zarel) [inline commands.js]: If you're not going to add `Ladders.visualizeAll` to `ladders-remote.js`, the next best way is to just return an error message here if `Ladders.visualizeAll` doesn't exist.

**Comment by bumbadadabum** [comment]
ok

**Comment by Zarel** [comment]
>branch name is mafia3

**Pull request merged**